### PR TITLE
@tus/s3-store: fix part number increment

### LINKED ISSUE #686: S3 Store stopped after uploading first chunk
### Initial checklist

- [X] I understand this is a bug report and questions should be posted in the [Community Forum](https://community.transloadit.com/c/tus/6)
- [X] I searched [issues](https://github.com/tus/tus-node-server/issues?q=is%3Aissue) and couldn’t find anything (or linked relevant results below)

### Steps to reproduce

Size of the video file on disk is 158MB, but minio shows ~70MB (that's my part size).
image

```js
// config/tus.js
const path = require("path");
const { Server, EVENTS } = require("@tus/server");
const { S3Store } = require("@tus/s3-store");

const env = require("./env");
const { s3ClientConfig } = require("./s3");

const s3Store = new S3Store({
  partSize: 80 * 1024 * 1024, // each uploaded part will have ~80MiB,
  s3ClientConfig: {
    ...s3ClientConfig,
    bucket: env.UPLOAD_BUCKET,
  },
});

const fileStore = new FileStore({
  directory: "./uploads",
});

const tusServer = new Server({
  path: "/upload",
  datastore: s3Store,
});

tusServer.on(EVENTS.EVENT_UPLOAD_COMPLETE, (event) => {
  console.log(
    `[${new Date().toLocaleTimeString()}] [EVENT HOOK] Upload complete for file ${
      event.file.id
    }`
  );
});

module.exports = tusServer;

```

```js
// app.js
const express = require("express");
const cors  = require("cors");
const env = require("./config/env");
const { initializeBuckets } = require("./config/s3");
const tusServer = require("./config/tus");

const app = express();
const uploadApp = express();

const PORT = env.PORT;

// app.use(express.static("public"));

app.use(cors());
uploadApp.all("*", tusServer.handle.bind(tusServer));
app.use("/upload", uploadApp);

async function startServer() {
  await initializeBuckets();

  app.listen(PORT, () => {
    console.log(`Server running on  http://localhost:${PORT}`);
  });
}

startServer();
```

Frontend code

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Veed - Video transcoding pipeline</title>
    <link
      href="https://releases.transloadit.com/uppy/v4.7.0/uppy.min.css"
      rel="stylesheet"
    />
    <style>
      .uppy-size--md .uppy-Dashboard-AddFiles-info {
        display: none;
      }
    </style>
  </head>
  <body>
    <h1>Veed - Video transcoding pipeline</h1>
    <noscript>You need JavaScript enabled for this example to work.</noscript>
    <button id="uppyModalOpener">Open Modal</button>
    <script type="module">
      import {
        Uppy,
        Dashboard,
        Webcam,
        Tus,
      } from "https://releases.transloadit.com/uppy/v4.8.0/uppy.min.mjs";

      const uppy = new Uppy({
        debug: true,
        autoProceed: false,
        restrictions: {
          maxFileSize: 1024 * 1024 * 1024, // 1GB
          allowedFileTypes: ["video/*"],
        },
      })
        .use(Dashboard, { trigger: "#uppyModalOpener" })
        .use(Webcam, { target: Dashboard })
        // .use(Tus, { endpoint: "https://tusd.tusdemo.net/files/" });
        .use(Tus, { endpoint: "http://localhost:8001/upload" });

      uppy.on("success", (fileCount) => {
        console.log(`${fileCount} files uploaded`);
      });
    </script>
  </body>
</html>

```

### Expected behavior

It should ideally upload the entire file.

### Actual behavior

Uploading just the first chunk of the video when s3 store is used. This works fine if s3 store is replaced with file store.

### LINKED ISSUE #688: Part number increments by 2 causing 10 000 limit to be hit sooner, and wrong part numbers
### Initial checklist

- [X] I understand this is a bug report and questions should be posted in the [Community Forum](https://community.transloadit.com/c/tus/6)
- [X] I searched [issues](https://github.com/tus/tus-node-server/issues?q=is%3Aissue) and couldn’t find anything (or linked relevant results below)

### Steps to reproduce

1. Upload a file of larger size such as 200GB
2. Choose an appropriate part size (32MB)
3. Enable logs so that part number is shown after each upload

### Expected behavior

The upload should complete without issue as 32MB parts is sufficient for 200GB without hitting the 10000 part limit.

### Actual behavior

Before we finish the upload we will get an error "InvalidArgument: Part number must be an integer between 1 and 10000, inclusive" from S3 because our part number is 10002.

Looking through the code I can see that the partNumber is actually incremented twice, which is why we hit the part number limit early, and why it complains about 10002.

First place we increment the part
https://github.com/tus/tus-node-server/blob/d0765dad65f09ca8dbbb3c7032e4635c2e615c23/packages/s3-store/src/index.ts#L567

Then before we upload the chunk we increment the part number again
https://github.com/tus/tus-node-server/blob/d0765dad65f09ca8dbbb3c7032e4635c2e615c23/packages/s3-store/src/index.ts#L370

### PR DESCRIPTION
**Pull request by fenos**
Fixes #686 
Fixes #688 

Caused by a [regression](https://github.com/tus/tus-node-server/commit/227ae813e0d550f5b413af42920b8f7c47695d47#diff-fecfad9cf273825148c4be6b30379a6331608acd8b71f5acd9fc60c2286be034L365-R370)

This PR correctly increment the part number

### CODE REVIEW

**Comment by fenos** [comment]
@Murderlon feel free to take over this PR

**Comment by Murderlon** [comment]
@fenos according to the test I added, your fix wasn't enough and I had to include 4f97d1a. If you have time give it a quick look, otherwise I'll merge it myself :)

File: .changeset/plenty-actors-compare.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@tus/s3-store": patch
+---
+
+Fix double increment for part numbers
```
Comment (Murderlon) [inline .changeset/plenty-actors-compare.md:5]:
```suggestion
Fix increment for part numbers
```

**Review by Murderlon — APPROVED** [review]
Thanks for the PR. Added a test to confirm.

Also I can't believe a change of formatters caused this regression but honestly a test should have caught it 🤔

**Comment by fenos** [comment]
Thanks a lot @Murderlon 🙏

**Pull request merged**